Makefile

```diff
@@ -135,13 +135,12 @@ YUI_COMPRESSOR ?= yuicompressor-2.4.8.jar
 HTML_PATH = $(abspath ./html)/
 WIFI_PATH = $(HTML_PATH)wifi/
 
-ESP_FLASH_MAX       ?= 503808  # max bin file
-
 ifeq ("$(FLASH_SIZE)","512KB")
 # Winbond 25Q40 512KB flash, typ for esp-01 thru esp-11
 ESP_SPI_SIZE        ?= 0       # 0->512KB (256KB+256KB)
 ESP_FLASH_MODE      ?= 0       # 0->QIO
 ESP_FLASH_FREQ_DIV  ?= 0       # 0->40Mhz
+ESP_FLASH_MAX       ?= 241664  # max bin file for 512KB flash: 236KB 
```

Comment:
Since this is a makefile (therefore it must account for all possibilities), someone may decide that not all provided features are needed for him - so he can purposedly exclude e.g. MQTT or SYSLOG or whatever. And in such cases the code will be smaller and **may** fit into smaller space. The logic here is that the size of the flash setting depends on the chip chosen - and this choice is in the hands of the user. Instead of fixing the number, would not it be better to make it dependable of the user choice ? All other flash parameters are already dependable, why not this one ?